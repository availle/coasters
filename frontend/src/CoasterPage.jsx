import React from 'react'
import { observer } from 'mobx-react'

@observer
export default class CoasterPage extends React.Component {
    componentDidMount() {
         this.props.coasterStore.get()
    }

    render() {
        const store = this.props.coasterStore;
        return (
            <div>
            { store.loading ? (
                <span>Loading...</span>
            ) : (
                <table>
                    <thead>
                    <tr>
                        <td>Name</td>
                        <td>Waiting time</td>
                    </tr>
                    </thead>
                    <tbody>
                    {this.renderCoasterRows(store)}
                    </tbody>
                </table>
            )}
            </div>)
    }

    renderCoasterRows(coasterStore) {
        return coasterStore.coasters.map((coaster, index) => {
            return <tr className='coaster' key={index}>
                <td>{coaster.name}</td>
                <td>{coaster.waitTime}</td>
            </tr>
        })
    }
}