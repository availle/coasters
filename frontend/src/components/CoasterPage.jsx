import React from 'react'
import { observer } from 'mobx-react'

import Coaster from './Coaster'

@observer
export default class CoasterPage extends React.Component {
    componentDidMount() {
        this.props.coasterStore.get()
    }

    render() {
        const store = this.props.coasterStore;
        return (
            <div>
                {store.loading ? (
                    <span>Loading...</span>
                ) : (
                    <table>
                        <thead>
                        <tr>
                            <td>Name</td>
                            <td>Waiting time</td>
                            <td>Thrill</td>
                        </tr>
                        </thead>
                        <tbody>
                        {this.renderCoasterRows(store)}
                        </tbody>
                    </table>
                )}
            </div>)
    }

    renderCoasterRows(coasterStore) {
        return coasterStore.coasters.map((coaster, index) => {
            return <Coaster name={coaster.name}
                            waitTime={coaster.waitTime}
                            key={index}
            />
        })
    }
}