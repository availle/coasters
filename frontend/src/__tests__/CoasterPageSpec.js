import React from 'react';
import { shallow } from 'enzyme'

import CoasterPage from '../components/CoasterPage'

describe('<CoasterPage/>', function () {
    let coasterStoreMock

    beforeEach(() => {
        coasterStoreMock = {
            loading: false,
            coasters: [{ name: 'test', waitTime: 3, thrill: 4}],
            get: () => {
            }
        }
    })

    it('should display coasters', function () {
        const renderedCoasterPage = shallow(<CoasterPage coasterStore={coasterStoreMock}/>);

        expect(renderedCoasterPage).toMatchSnapshot()
    });

    it('should display "Loading..." if loading is true', () => {
        coasterStoreMock.loading = true
        const renderedCoasterPage = shallow(<CoasterPage coasterStore={coasterStoreMock}/>);
        expect(renderedCoasterPage.find('span').text()).toEqual('Loading...')
    })
});