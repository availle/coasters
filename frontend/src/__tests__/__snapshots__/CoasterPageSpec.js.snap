// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CoasterPage/> should display coasters 1`] = `
<div>
  <table>
    <thead>
      <tr>
        <td>
          Name
        </td>
        <td>
          Waiting time
        </td>
      </tr>
    </thead>
    <tbody>
      <Coaster
        key="0"
        name="test"
        waitTime={3}
      />
    </tbody>
  </table>
</div>
`;
